pkg/core/native/native_neo.go

```diff
@@ -527,19 +505,22 @@ func (n *NEO) distributeGas(ic *interop.Context, acc *state.NEOBalance) (*big.In
 	if ic.Block == nil || ic.Block.Index == 0 || ic.Block.Index == acc.BalanceHeight {
 		return nil, nil
 	}
-	gen, err := n.calculateBonus(ic.DAO, acc.VoteTo, &acc.Balance, acc.BalanceHeight, ic.Block.Index)
+	gen, err := n.calculateBonus(ic, acc)
 	if err != nil {
 		return nil, err
 	}
 	acc.BalanceHeight = ic.Block.Index
+	if acc.VoteTo != nil {
+		latestGasPerVote := n.getLatestGASPerVote(ic.DAO, makeVoterKey(acc.VoteTo.Bytes()))
+		acc.LastGasPerVote = latestGasPerVote
+	}
 
 	return gen, nil
 }
 
 func (n *NEO) unclaimedGas(ic *interop.Context, args []stackitem.Item) stackitem.Item {
 	u := toUint160(args[0])
-	end := uint32(toBigInt(args[1]).Int64())
```

Comment:
BTW, this changes the interface for `unclaimedGas` Neo contract's method, we can't do it this easy (although I doubt there are any uses of it). At the same time it only makes sense when `end == BlockHeight()+1`, values lower than that can't in fact be claimed, values higher than that are not known yet exactly for voters.

Reply:
> this changes the interface for `unclaimedGas`

Yeah. Return error when `end != BlockHeight+1`?